src/swoop/api/routers/payloads.py

```diff
@@ -42,13 +111,52 @@ def list_payloads(
         "500": {"model": APIException},
     },
 )
-def get_job_status(
-    payload_id: str = Path(..., alias="payloadId"),
+async def get_payload_status(
+    request: Request, payload_id
 ) -> PayloadInfo | APIException:
     """
     retrieve info for a payload
     """
-    pass
+
+    async with request.app.state.readpool.acquire() as conn:
+        q, p = render(
+            """
+                SELECT
+                c.payload_uuid, payload_hash, workflow_version, workflow_name,
+                c.created_at, invalid_after, item_id, collection, action_uuid
+                FROM swoop.payload_cache c
+                INNER JOIN swoop.item_payload p
+                ON c.payload_uuid = p.payload_uuid
+                INNER JOIN swoop.action a
+                ON c.payload_uuid = a.payload_uuid
+                INNER JOIN swoop.input_item i
+                ON p.item_uuid = i.item_uuid
+                WHERE c.payload_uuid = :payload_id::uuid;
```

Comment:
What you are returning here is a bit nonsensical, in that you get a row for _every_ item in the payload, and _every_ job that uses said payload, combinatorially.

That is, if you have a payload with 100 items in it (not common, but not unheard of) with 10 jobs, you end up with 1000 rows returned. That a ton of data to transfer from the DB to the application, then deserialized into python objects (an expensive process), then thrown away because it is useless.

If you wanted to keep this to a single query then return arrays of items and action UUIDs. Except that items are composed of two values--arrays really only work in this context with a single value, otherwise we have to build json or structured strings of some other sort and things get messy fast. If we remove the items from the the query though, we can use the array idea for jobs:

```
SELECT
  c.payload_uuid,
  c.payload_hash,
  c.workflow_version,
  c.workflow_name,
  c.created_at,
  c.invalid_after,
  array(
    SELECT
      action_uuid
    FROM swoop.action
    WHERE
      payload_uuid = c.payload_uuid
  ) AS action_uuids
FROM swoop.payload_cache AS c
WHERE
  c.payload_uuid = :payload_id::uuid
```

Then you can simply make another query for the items like:

```
SELECT
  item_id,
  collection
FROM swoop.item_payload AS ip
INNER JOIN swoop.input_item USING (item_uuid)
WHERE
  ip.payload_uuid = :payload_id::uuid
```